### ISSUE TITLE
localstorage was cleared, when i restart my sketch

### ISSUE THREAD

**Issue by zobor** [skpm/sketch-module-web-view#78]
can you help me?

**Comment by mathieudutour** [comment]
Yeah I don’t think there is another solution. That’s also what I’m doing in the Sketch DevTools.

 What we could do perhaps is automate this by monkey-patching localStorage automatically so that when you can `localStorage`, it would actually call the plugin’s “backend”

**Comment by happylinks** [comment]
That would be nice! Maybe with a special prefix so it’ll never conflict.

**Comment by mathieudutour** [comment]
Yes exactly. Do you want to have a go at it? There is already an example of monkey patching `window.postMessage`, it should probably be similar

**Comment by happylinks** [comment]
Sure, I can probably have a go at it this weekend or next week👌

**Comment by mathieudutour** [comment]
Perfect! Give me a shout if you need something

**Comment by zobor** [comment]
> Same. What I did to "solve" it, is persist it to the sketch plugin settings.
> https://developer.sketchapp.com/reference/api/#set-a-plugin-setting
> 
> In the sketch "backend" code I used:
> 
> ```
>  const storedActiveProjectID = sketch.Settings.settingForKey(
>         'storedActiveProjectID'
>       );
>       if (storedActiveProjectID) {
>         browserWindow.webContents.executeJavaScript(
>           `localStorage.setItem('storedActiveProjectID', "${storedActiveProjectID}")`
>         );
>       }
> ```
> And in the frontend I used a pluginCall to pass it to the backend on change.
> 
> Not the greatest solution, but maybe it helps you!

thanks, although it can't solve all the problems, but helpful.
